toolchest_client/tools/api.py

```diff
@@ -628,6 +628,76 @@ def rapsearch2(inputs, output_path=None, output_primary_name="output", database_
 rapsearch = rapsearch2
 
 
+def salmon(output_path=None, tool_args="", read_one=None, read_two=None, single_end=None, library_type="A",
+           database_name="salmon_hg38", database_version="1", **kwargs):
+    """Runs Salmon via Toolchest.
+
+    :param database_name: Name of database to use for STAR alignment (defaults to GRCh38).
+    :param database_version: Version of database to use for STAR alignment (defaults to 1).
+    :param library_type: (optional) `--libType` value. Defaults to "A" for automatic. See
+    https://salmon.readthedocs.io/en/latest/salmon.html#what-s-this-libtype
+    :param output_path: (optional) Path (local or S3) to a directory where the output files will be downloaded.
+    :param read_one: (optional) `-1` inputs. Path or list of paths for read 1 of paired-read input files.
+    :param read_two: (optional) `-2` inputs. Path or list of paths for read 2 of paired-read input files.
+    :param single_end: (optional) `-r` inputs. Path or list of paths for single-end inputs (no interleaved files).
+    :param tool_args: (optional) Additional arguments to be passed to Salmon.
+
+    .. note:: Each read in `read_one` should match with a read in `read_two`, and vice
+    versa. In other words, the nth read in `read_one` should be paired with the nth read
+    in `read_two`.
+
+    Usage::
+
+        >>> import toolchest_client as toolchest
+        >>> toolchest.salmon(
+        ...     tool_args="",
+        ...     read_one=["./pair_1/r1.fa", "./pair_2/r1.fa"],
+        ...     read_two=["./pair_1/r2.fa", "./pair_2/r2.fa"],
+        ...     output_path="./path/to/output",
+        ... )
+
+    """
+
+    # Guard against library type in tool args
+    if "-l" in tool_args or "--libType" in tool_args:
+        raise ValueError("A library type is set in 'tool_args'. Use the 'library_type' argument instead.")
+
+    # If input parameters are lists, parse these for input_prefix_mapping.
+    tag_to_param_map = {
+        "-1": read_one,
+        "-2": read_two,
+        "-r": single_end,
+    }
+    input_list = []  # list of all inputs
+    input_prefix_mapping = {}  # map of each input to its respective tag
+    for tag, param in tag_to_param_map.items():
+        if isinstance(param, list):
+            for index, input_file in enumerate(param):
+                input_list.append(input_file)
+                input_prefix_mapping[input_file] = {
+                    "prefix": tag,
+                    "order": index,
+                }
+        elif isinstance(param, str):
+            input_list.append(param)
+            input_prefix_mapping[param] = {
+                "prefix": tag,
+                "order": 0,
+            }
```

Comment:
This is nearly identical to what we do with `megahit`, goal is to preserve tool arg ordering